src/process/io/s3/FQIS3Client.h

```diff
@@ -0,0 +1,230 @@
+            }
+            result = tuple<bool, std::list<ObjectListEntry>>(outcome.IsSuccess(), entries);
+            return outcome;
+        });
+        return result;
+    }
+
+    FQIS3ClientRequestBooleanResult checkObjectExistence() {
+        auto list = getObjectList();
+        bool found = false;
+        if (std::get<0>(list)) {
+            for (const auto &entry : std::get<1>(list)) {
+                if (entry.name == object_name) {
+                    found = true;
+                    break;
+                }
+            }
+        }
+        return FQIS3ClientRequestBooleanResult(std::get<0>(list), found);
+    }
+
+    /**
+     * Returns the object size for this clients object.
+     * @return A tuple indicating [success, size]
+     */
+    tuple<bool, int64_t> getObjectSize() {
+        auto[success, list] = getObjectList();
+        bool found = false;
+        int64_t size = 0;
+        if (!success)
+            return {false, 0};
+
+        for (const auto& entry : list) {
+            if (entry.name == object_name) {
+                found = true;
+                size = entry.size;
+                break;
+            }
+        }
+        return {found, size};
+    }
+
+    FQIS3ClientRequestBooleanResult putFile(const string &file) {
+        bool couldPut = false;
+
+        bool requestResult = performS3Request<PutObjectOutcome>([=](S3Client &client) -> PutObjectOutcome {
+            PutObjectRequest object_request;
+            object_request.SetBucket(bucket_name.c_str());
+            object_request.SetKey(object_name.c_str());
+            auto input_data =
+                    Aws::MakeShared<FStream>("FASTQ index", file.c_str(), std::ios_base::in | std::ios_base::binary);
+            object_request.SetBody(input_data);
+            object_request.SetContentMD5(
+                    HashingUtils::Base64Encode(HashingUtils::CalculateMD5(*object_request.GetBody())));
+            return client.PutObject(object_request);
+        });
+
+        return FQIS3ClientRequestBooleanResult(requestResult, requestResult);
+    }
+
+    /**
+     * As I had big memory issues when using the same S3 stream for the whole index / extract process, I now try to read
+     * things chunk-wise. This might not be the most efficient way, but the memory consumption was around 1.2 times as
+     * as the file size. Dealing with larger files (think of 30 or 40GB) is simply not possible.
+     * @param position From where to take
+     * @param length   How much to take
+     * @param buffer   Where to store to. This buffer needs to be large enough to hold the data!
+     * @return A tuple with a success indicator and the amount of read Bytes.
+     */
+    tuple<bool, int64_t> readBlockOfData(int64_t position, int64_t length, Bytef *buffer) {
+        int64_t readBytes{0};
+        bool success = performS3Request<GetObjectOutcome>([&](S3Client &client) -> GetObjectOutcome {
+            GetObjectRequest object_request;
```

Comment:
I understood that you want to prefer camel-case over snake-case ... ?

Reply:
copy&paste :(